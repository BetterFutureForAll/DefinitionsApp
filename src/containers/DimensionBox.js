import React from "react";
import data from '../assets/definitions.json';
const keyFixer = (key) => key.replace(/[ \t]+$/, '').replace(/ /g, "_");;

// export const nestedData = (data) => {
//   let keys = ['dimension', 'component', 'indicator_name']
//   let result = data.reduce((r, o) => {
//     let key;
//     keys.reduce((t, k) => {
//       ({ [k]: key, ...o } = o);
//       if (!t[key]) {
//         key = keyFixer(key);
//         console.log('t:', t, 'key:', key);
//         t[key] = { _: [] };
//         t._.push({ [k]: key, children: t[key]._ });
//         console.log('t._:', t._);
//       }
//       return t[key];
//     }, r)
//       ._
//       .push(o);
//   }, { _: [] })
//     ._;
//   return result;
// };

