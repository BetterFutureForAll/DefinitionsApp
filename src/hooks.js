import { useState } from 'react';

export const regEx = (d) => {
  return d.replace(/ /g, "_");
}

export const groupBy = (objectArray, property) => {
  let result = objectArray.reduce((acc, obj) => {
    const key = obj[property];
    const curGroup = acc[key] ?? [];
    return { ...acc, [key]: [...curGroup, obj] };
  }, {});
  // filter out blank fields 
  // mutates object
  function clean(obj) {
    for (var propName in obj) {
      if (propName === undefined || propName === 'undefined') {
        delete obj[propName];
      }
    }
    return obj;
  }

  return clean(result);
};

export const useActive = (initialState) => {
  const [activeValue, setActiveValue] = useState(initialState);
  const toggler = () => { setActiveValue(!activeValue) };
  return [activeValue, toggler]
};