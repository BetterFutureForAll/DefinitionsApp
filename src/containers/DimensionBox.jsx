import React from "react";
import data from '../assets/definitions.json';
import { groupBy } from "../hooks";
const keyFixer = (key) => key.replace(/[ \t]+$/, '').replace(/ /g, "_");;

const Dimensions = () => {
let dimensionsData = groupBy(data, 'dimension')
//return arrays of unique dimensions datasets

}

export default Dimensions;


