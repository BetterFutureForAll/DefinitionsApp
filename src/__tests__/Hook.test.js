import { parsedDef, regEx } from "../hooks";
import rawData from '../assets/definitions.json';
import { nestedData } from '../DimensionBox';
import { useParsedData } from '../hooks';
let data = rawData.definitionsArray;

describe('test suite for hooks functions', ()=>  {

  test('regEx returns string without whitespace', async () => {
    let string = 'abc d';
    let expected = 'abc_d';
    let result = regEx(string);
    expect(result).toBe(expected);
  });
  
  //skipped due to CSV path parsing error is JSDOM
  test('should parse a CSV and return an array of objects', async () => {
    let [data, setData] = useParsedData();
    let result = await data;
    let objectMock = { "key": "value" };
    await expect(result).resolves.toBe(objectMock);
    await expect(result).rejects.toMatch('error');
  
  });

  test('should return a nested data set by keys', () => {
    console.log(nestedData(data));
    expect(nestedData).toMatchSnapshot();
  });
});