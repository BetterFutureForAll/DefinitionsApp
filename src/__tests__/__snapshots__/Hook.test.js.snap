// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite for hooks functions should return a nested data set by keys 1`] = `[Function]`;
